go/enclave/evm/evm_facade.go

```diff
@@ -186,3 +273,33 @@ func createCleanState(s *state.StateDB, msg *gethcore.Message, ethHeader *types.
 	cleanState.Prepare(chainConfig.Rules(ethHeader.Number, true, 0), msg.From, ethHeader.Coinbase, msg.To, nil, msg.AccessList)
 	return cleanState
 }
+
+func (exec *evmExecutor) chargeBaseFeeOnly(s *state.StateDB, header *types.Header, payer gethcommon.Address, gasUsed uint64) {
+	if gasUsed == 0 || header.BaseFee == nil || header.BaseFee.Sign() == 0 {
+		return
+	}
+	weiBig := new(big.Int).Mul(new(big.Int).SetUint64(gasUsed), header.BaseFee)
+	if weiBig.Sign() <= 0 {
+		return
+	}
+	wei, _ := uint256.FromBig(weiBig)
+	s.SubBalance(payer, wei, tracing.BalanceDecreaseGasBuy)
+	s.AddBalance(header.Coinbase, wei, tracing.BalanceIncreaseRewardTransactionFee)
+}
+
+// readVisibilityWithCap reserves cap gas from the GasPool, invokes the visibility reader,
+// and refunds any unused portion back to the pool. On error, it refunds the full cap if reserved.
+func (exec *evmExecutor) readVisibilityWithCap(ctx context.Context, evmEnv *vm.EVM, gp *gethcore.GasPool, addr gethcommon.Address, cap uint64) (*core.ContractVisibilityConfig, uint64, error) {
+	if err := gp.SubGas(cap); err != nil {
```

Comment:
I don't quite understand this gasPool logic. Why is it needed?

Reply:
I might have gotten confused, need to investigate if geth auto handles it when we give the gas pool .. its basically the pool for the batch we have, but not sure if its proper handled when doing a static call on the evm, rather than applying transactions.

Reply:
the static call does not handle it - thus you need to manually do it and ensure gas has been spent.